src/roles/util.ts

```diff
@@ -160,7 +187,11 @@ class PermissionsMapping {
   public static use = (role: WeaviateRole) => new PermissionsMapping(role);
 
   public map = (): Role => {
-    this.role.permissions.forEach(this.permissionFromWeaviate);
+    // If truncated roles are requested (?includeFullRoles=false),
+    // role.permissions are not present.
+    if (this.role.permissions !== null) {
+      this.role.permissions.forEach(this.permissionFromWeaviate);
+    }
```

Comment:
I think, because of how the OpenAPI schema is defined, this clause won't fire. I think you should add a test of `.getAssignedRoles` with `includePermissions: true` and see what happens. But, given the Python implementation, I think you will ned to pas `inlcudePermissions` here to the `map` function and use whether that is true or not to parse the `this.role.permissions`. Either way, I think it is returned as an empty list even if `includePermissions: false`

Reply:
Let me check again -- I am rather positive that I've only added this clause because a test I wrote for `userAssignments` was initially failing.

Reply:
Getting this error if the if-clause is removed:

(in the test context)
```sh
Rejected to value: [TypeError: Cannot read properties of null (reading 'forEach')]
```

I'll add that test to the suite to illustrate.

Reply:
Okay, very interesting! Thanks for verifying 😁